### ISSUE TITLE
No need to use nn.Softmax layer when using nn.CrossEntropyLoss()

### ISSUE THREAD

**Issue by jshin49** [IBM/dstc7-noesis#18]
In this line
```python
output = self.final_layer(logits)
```
https://github.com/IBM/dstc7-noesis/blob/b0f2d6009b1e4519509843267e5fb3091242a879/noesis/networks/dual_encoder.py#L196

Since you are using nn.CrossEntropyLoss() as the default loss function in 
```python
def __init__(self, loss_func=CrossEntropyLoss(), batch_size=64):
```
https://github.com/IBM/dstc7-noesis/blob/b0f2d6009b1e4519509843267e5fb3091242a879/noesis/evaluator/evaluator.py#L16

which combines nn.LogSoftmax() and nn.NLLLoss(), according to [PyTorch docs](https://pytorch.org/docs/stable/nn.html#crossentropyloss), there is no need to use a final softmax output layer.

Hence change it to
```python
output = logits
```
or just return logits.

**Comment by kpfadnis** [comment]
@jshin49 You are correct. Like you pointed out `CrossEntropyLoss` already takes care of it. 

Will fix in next release.
